protocol/statetracker/state_tracker.go

```diff
@@ -40,6 +43,27 @@ type Updater interface {
 	UpdaterKey() string
 }
 
+type SpecUpdaterInf interface {
+	RegisterForSpecUpdates(ctx context.Context, specUpdatable updaters.SpecUpdatable, endpoint lavasession.RPCEndpoint) error
+}
+
+// Either register for spec updates or set spec for offline spec, used in both consumer and provider process
+func RegisterForSpecUpdatesOrSetStaticSpec(ctx context.Context, chainParser chainlib.ChainParser, specPath string, rpcEndpoint lavasession.RPCEndpoint, specUpdaterInf SpecUpdaterInf) (err error) {
```

Comment:
i wouldn't put this function here, but put in in common instead because it's not related to statetracker when there's a static spec